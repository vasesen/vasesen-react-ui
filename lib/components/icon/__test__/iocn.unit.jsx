import * as renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../icon'
import {mount} from 'enzyme'

describe('Icon',()=>{
    it('svg',()=>{
        const svgJson = renderer.create(<Icon name="footer"/>).toJSON()
        expect(svgJson).toMatchSnapshot() //快照
    })
    
     it('onCkick',()=>{
         const fn = jest.fn()
         const component = mount(<Icon name="love" onClick={fn}></Icon>)
         component.find('svg').simulate('click')
         expect(fn).toBeCalled()
     })
})