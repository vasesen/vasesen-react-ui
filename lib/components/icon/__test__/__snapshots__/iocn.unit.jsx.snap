// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon svg 1`] = `
<svg
  className="va-icon"
>
  <use
    xlinkHref="#footer"
  />
</svg>
`;
